#Marker Components:

# Model Groups
wheel_marker: &wheel_marker
  - [wheel0, 0, [0.65, 0.3, 0.15]]
  - [wheel1, 1, [0.65, -0.3, 0.15]]
  - [wheel2, 2, [-0.65, -0.3, 0.15]]
  - [wheel3, 3, [-0.65, 0.3, 0.15]]
tower_marker: &tower_marker
  - [tower0, 4, [0.6, 0.3, 0.45]]
  - [tower1, 5, [0.6, -0.3, 0.45]]
  - [tower2, 6, [-0.6, -0.3, 0.45]]
  - [tower3, 7, [-0.6, 0.3, 0.45]]
front_back_pipe_marker: &front_back_pipe_marker
  - [pipe0, 8, [0.0, 0.32, 0.5]]
  - [pipe1, 9, [0.0, 0.32, 0.4]]
  - [pipe2, 10, [0.0, -0.32, 0.5]]
  - [pipe3, 11, [0.0, -0.32, 0.4]]
left_right_pipe_marker: &left_right_pipe_marker
  - [pipe4, 12, [0.65, 0.0, 0.5]]
  - [pipe5, 13, [0.65, 0.0, 0.4]]
  - [pipe6, 14, [-0.65, 0.0, 0.5]]
  - [pipe7, 15, [-0.65, 0.0, 0.4]]
name_marker: &name_marker
  - [name, 16, [0, 0, 0.6]]
bed_marker: &bed_marker
  - [bed, 17, [0,0,0.575]]


hunter_robot:
  wheel_marker:
    - [wheel0, 0, [0.4, 0.2, 0.15]]
    - [wheel1, 1, [0.4, -0.2, 0.15]]
    - [wheel2, 2, [-0.4, -0.2, 0.15]]
    - [wheel3, 3, [-0.4, 0.2, 0.15]]
  name_marker:
    - [name, 4, [0, 0, 0.25]]
  bed_marker:
    - [bed, 5, [0,0,0.31]]



# Agent Models
short_robot: &short_robot
  wheel_marker: *wheel_marker
  tower_marker: *tower_marker
  front_back_pipe_marker: *front_back_pipe_marker
  left_right_pipe_marker: *left_right_pipe_marker
  name_marker: *name_marker
  bed_marker: *bed_marker
short_transportation: *short_robot
short_sprayer: *short_robot


short_robot_load_1:
  wheel_marker: *wheel_marker
  tower_marker: *tower_marker
  front_back_pipe_marker: *front_back_pipe_marker
  left_right_pipe_marker: *left_right_pipe_marker
  name_marker: *name_marker
  bed_marker: *bed_marker
  empty_tray_marker:
    - [etray1, 18, [0.25, 0, 0.65]]
  full_tray_marker:
    - [ftray1, 19, [0.25, 0, 0.7]]


short_robot_load_2:
  wheel_marker: *wheel_marker
  tower_marker: *tower_marker
  front_back_pipe_marker: *front_back_pipe_marker
  left_right_pipe_marker: *left_right_pipe_marker
  name_marker: *name_marker
  bed_marker: *bed_marker
  empty_tray_marker:
    - [etray1, 18, [0.25, 0, 0.65]]
    - [etray2, 19, [-0.25, 0, 0.65]]
  full_tray_marker:
    - [ftray1, 20, [0.25, 0, 0.7]]
    - [ftray2, 21, [-0.25, 0, 0.7]]


short_robot_load_3:
  wheel_marker: *wheel_marker
  tower_marker: *tower_marker
  front_back_pipe_marker: *front_back_pipe_marker
  left_right_pipe_marker: *left_right_pipe_marker
  name_marker: *name_marker
  bed_marker: *bed_marker
  empty_tray_marker:
    - [etray1, 18, [0.25, 0, 0.65]]
    - [etray2, 19, [-0.25, 0, 0.65]]
    - [etray3, 20, [0.25, 0, 0.85]]
  full_tray_marker:
    - [ftray1, 21, [0.25, 0, 0.7]]
    - [ftray2, 22, [-0.25, 0, 0.7]]
    - [ftray3, 23, [0.25, 0, 0.9]]


short_robot_load_4:
  wheel_marker: *wheel_marker
  tower_marker: *tower_marker
  front_back_pipe_marker: *front_back_pipe_marker
  left_right_pipe_marker: *left_right_pipe_marker
  name_marker: *name_marker
  bed_marker: *bed_marker
  empty_tray_marker:
    - [etray1, 18, [0.25, 0, 0.65]]
    - [etray2, 19, [-0.25, 0, 0.65]]
    - [etray3, 20, [0.25, 0, 0.85]]
    - [etray4, 21, [-0.25, 0, 0.85]]
  full_tray_marker:
    - [ftray1, 22, [0.25, 0, 0.7]]
    - [ftray2, 23, [-0.25, 0, 0.7]]
    - [ftray3, 24, [0.25, 0, 0.9]]
    - [ftray4, 25, [-0.25, 0, 0.9]]




tall_robot: &tall_robot
  wheel_marker: *wheel_marker
  tower_marker: *tower_marker
  front_back_pipe_marker: *front_back_pipe_marker
  name_marker: *name_marker
  arch_marker:
    - [arch, 12, [-0.68, -0.37, 0.4]]
tall_data_collection: *tall_robot
tall_sprayer: *tall_robot
tall_uv: *tall_robot


human: &human
  leg_marker:
    - [left_leg, 0, [0, 0.05, 0.4]]
    - [right_leg, 1, [0, -0.05, 0.4]]
  body_marker:
    - [body, 2, [0, 0, 1.05]]
  arm_marker:
    - [left_arm, 3, [0, 0.15, 1.0]]
    - [right_arm, 4, [0, -0.15, 1.0]]
  head_marker:
    - [head, 5, [0, 0, 1.40]]
  name_marker:
    - [name, 6, [0, 0, 1.8]]
picker: *human

disc: &disc
  node_marker:
    - [flat, 0, [0, 0, 0.01]]
  name_marker:
    - [name, 2, [0, 0, 0.5]]
special_node: *disc
